src/app/shared/components/addregressionregion/addregressionregion.component.ts

```diff
@@ -339,60 +340,70 @@ export class AddRegressionRegionModal implements OnInit, OnDestroy {
+              this.cancelCreateRegression();
             }
-        }, error => {
+            this.requeryFilters();
+            this.cancelCreateRegression();
+          }, error => {
             this._loaderService.hideFullPageLoad();
-            if (this._settingsService.outputWimMessages(error)) {return; }
-            this._toasterService.pop('error', 'Error editing Regression Region', error.message || error._body.message || error.statusText); }
-        );  
+            if (this._settingsService.outputWimMessages(error)) { return; }
+            this._toasterService.pop('error', 'Error creating Citation', error.message || error._body.message || error.statusText);
+          }
+        );
+      } else if (this.addCitation && this.newCitation == true) { // New citation
+        this.createNewCitation(this.selectedRegRegion);
+      } else if (this.removeCit == true) { // Remove citation
+        const idx = this.regressionRegions.findIndex(r => r.id === this.rr);
+        if (idx != -1) {
+          const regReg = this.regressionRegions[idx];
+          regReg.citationID = null;
+          this._settingsService.putEntity(this.rr, regReg, this.configSettings.regRegionURL)
+            .subscribe((response) => {
+              this.requeryFilters();
+              this._nssService.outputWimMessages(response);
+            }, error => {
+              if (this._settingsService.outputWimMessages(error)) {return; }
+              this._toasterService.pop('error', 'Error removing Citation', error._body.message || error.statusText);
+            }
+          );
+          this.selectedRegRegion.citationID = null;
+          this.cancelCreateRegression();
+        }
+      } else { // add existing citation
+        this.requeryFilters();
+        this.cancelCreateRegression();
+      }
+    }, error => {
+      this._loaderService.hideFullPageLoad();
+      if (this._settingsService.outputWimMessages(error)) { return; }
+      this._toasterService.pop('error', 'Error editing Regression Region', error.message || error._body.message || error.statusText); }
+    );  
   }
 
   public removeCitation(){
-    this.saveFilters();
-    const idx = this.regressionRegions.findIndex(r => r.id === this.rr);
-    const regReg = this.regressionRegions[idx];
-    regReg.citationID = null;
-    this._settingsService.putEntity(this.rr, regReg, this.configSettings.regRegionURL)
-        .subscribe((response) => {
-            this.requeryFilters();
-            this._nssService.outputWimMessages(response);
-        }, error => {
-            if (this._settingsService.outputWimMessages(error)) {return; }
-            this._toasterService.pop('error', 'Error removing Citation', error._body.message || error.statusText);
-        }
-    );
-    this.selectedRegRegion.citationID=null;
+    if (this.selectedRegRegion) { // if creating new regression region cannot set citationID to null
+      this.selectedRegRegion.citationID = null
```

Comment:
This needs to change the citation ID of the newRegReg form. Otherwise, it doesn't get sent through when saving the reg region.